### PR TITLE
jc - Add HTTP PUT for the `Commons` type.

### LINKED ISSUE #33: 6pm-4: Add PUT method to the commons controller.
- [x] The commons controller has an HTTP PUT method that creates a `Commons` if it does not exist (status 201).
- [x] The commons controller has an HTTP PUT method that edits the `Commons` if it does exist (status 204), locating the `Commons` using the identifier in the query parameter.
- [x] An HTTP PUT method should not have a response body.

### PR DESCRIPTION
**Pull request by jacksoncooper**
put

### CODE REVIEW

**Review by sytarno — DISMISSED** [review]
Tested it locally via swagger-ui. Everything works as expected, good job! Received 204's when trying to edit. Hopefully we don't mistakenly create a new one while trying to edit.

**Comment by jacksoncooper** [comment]
Issue #33.

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java [hunk 1]
```diff
@@ -57,6 +59,38 @@
...
+    Commons updated;
+    HttpStatus status;
+
+    if (existing.isPresent()) {
+      updated = existing.get();
+      status = HttpStatus.NO_CONTENT;
+    } else {
+      updated = new Commons();
+      status = HttpStatus.CREATED;
+    }
```
Comment (btk5h) [inline src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java:80]: Thank you for following HTTP status code conventions. I don't think we ever covered this explicitly in class, but the attention to detail is nice.

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java [hunk 2]
```diff
@@ -57,6 +59,38 @@
     return ResponseEntity.ok().body(body);
   }
 
+  @ApiOperation(value = "Update a commons")
+  @PreAuthorize("hasRole('ROLE_ADMIN')")
+  @PutMapping("/update")
+  public ResponseEntity<String> updateCommons(
+    @ApiParam("commons identifier") @RequestParam long id,
+    @ApiParam("request body") @RequestBody CreateCommonsParams params
+  ) throws JsonProcessingException
```
Comment (btk5h) [inline src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java]: We can remove `JsonProcessingException` from the method signature if the method cannot throw it.

**Review by btk5h — DISMISSED** [review]

File: src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java (hunk 2 above)
Comment (jacksoncooper) [reply src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java]: Oooh thank you I forgot I wasn't giving anything in the response body.

**Review by Jacqueskim — DISMISSED** [review]
LGTM

**Review by sytarno — DISMISSED** [review]
Good catch by Bryan. Everything still works as intended, so looks good to approve. Not much changed from my last review.

**Review by pconrad — DISMISSED** [review]
LGTM--- will want to test on Heroku before merging though

**Review by rsdharmaji — APPROVED** [review]

**Pull request merged**